src/brod_client.erl

```diff
@@ -520,21 +576,21 @@ start_metadata_socket(_ClientId, [], Reason) ->
 start_metadata_socket(ClientId, [Endpoint | Endpoints], _Reason) ->
   {Host, Port} = Endpoint,
   case brod_sock:start_link(self(), Host, Port, ClientId, []) of
-    {ok, Pid}       -> #sock{ endpoint = Endpoint
-                            , sock_pid = Pid
-                            };
+    {ok, Pid}       -> {ok, #sock{ endpoint = Endpoint
+                                 , sock_pid = Pid
+                                 }};
     {error, Reason} -> start_metadata_socket(ClientId, Endpoints, Reason)
   end.
 
-maybe_start_producers_sup([]) ->
-  {ok, undefined};
-maybe_start_producers_sup(Producers) ->
-  brod_producers_sup:start_link(self(), Producers).
-
-maybe_start_consumers_sup([]) ->
-  {ok, undefined};
-maybe_start_consumers_sup(Consumers) ->
-  brod_consumers_sup:start_link(self(), Consumers).
+-spec is_topic_exist(pid(), topic()) -> boolean().
+is_topic_exist(Socket, Topic) ->
+  Request = #kpro_MetadataRequest{topicName_L = []},
```

Comment:
why don't send just one topic name in the list ?
it can be a quite a lot of info for the broker to collect when trying to reply this request.